functions.cpp

```diff
@@ -760,6 +760,10 @@ namespace Sass {
     Signature unquote_sig = "unquote($string)";
     BUILT_IN(sass_unquote)
     {
+      if (dynamic_cast<Null*>(env["$string"])) {
+        return new (ctx.mem) Null(pstate);
+      }
+
       To_String to_string(&ctx);
       AST_Node* arg = env["$string"];
```

Comment:
Would make more sense IMO to use to `arg` variable and avoid fetching it twice from env ... maybe you want to move the `To_String` inside the if cases that actually uses it ...